api/api-config-repos-v1/src/main/java/com/thoughtworks/go/apiv1/configrepo/ConfigReposControllerV1.java

```diff
@@ -0,0 +1,109 @@
+import com.thoughtworks.go.config.remote.ConfigRepoConfig;
+import com.thoughtworks.go.domain.materials.MaterialConfig;
+import com.thoughtworks.go.server.materials.MaterialUpdateService;
+import com.thoughtworks.go.server.service.ConfigRepoService;
+import com.thoughtworks.go.server.service.MaterialConfigConverter;
+import com.thoughtworks.go.spark.Routes;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.util.Objects;
+
+import static spark.Spark.*;
+
+@Component
+public class ConfigReposControllerV1 extends ApiController implements SparkSpringController {
+
+    private final ApiAuthenticationHelper authenticationHelper;
+    private final GoRepoConfigDataSource goRepoConfigDataSource;
+    private final ConfigRepoService configRepoService;
+    private final MaterialUpdateService mus;
+    private final MaterialConfigConverter converter;
+
+    @Autowired
+    public ConfigReposControllerV1(ApiAuthenticationHelper authenticationHelper, GoRepoConfigDataSource goRepoConfigDataSource, ConfigRepoService configRepoService, MaterialUpdateService mus, MaterialConfigConverter converter) {
+        super(ApiVersion.v1);
+        this.authenticationHelper = authenticationHelper;
+        this.goRepoConfigDataSource = goRepoConfigDataSource;
+        this.configRepoService = configRepoService;
+        this.mus = mus;
+        this.converter = converter;
+    }
+
+    @Override
+    public String controllerBasePath() {
+        return Routes.ConfigRepos.INTERNAL_BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("", mimeType, authenticationHelper::checkAdminUserAnd403);
+            before("", this::verifyContentType);
+
+            before("/*", mimeType, this::setContentType);
+            before("/*", mimeType, authenticationHelper::checkAdminUserAnd403);
+            before("/*", this::verifyContentType);
+
+            get(Routes.ConfigRepos.LAST_PARSED_RESULT_PATH, mimeType, this::getLastParseResult);
+            post(Routes.ConfigRepos.TRIGGER_UPDATE_PATH, mimeType, this::triggerUpdate);
+        });
+    }
+
+    String getLastParseResult(Request req, Response res) {
+        ConfigRepoConfig repo = Objects.requireNonNull(getRepoFromRequest(req));
+
+        final PartialConfigParseResult result = goRepoConfigDataSource.getLastParseResult(repo.getMaterialConfig());
+
+        return PartialConfigParseResultRepresenter.toJSON(result);
+    }
+
+    private ConfigRepoConfig getRepoFromRequest(Request req) {
+        ConfigRepoConfig repo = configRepoService.getConfigRepo(req.params(":id"));
+
+        if (null == repo) {
+            HaltApiResponses.haltBecauseNotFound();
```

Comment:
Although calling `halt` itself throws, without an explicit `throw` there's no way to know if the stack unrolls itself. Change to `throw HaltApiResponses.something()` please :)